### PR TITLE
[improvement] Implement capability to assign FIDP groups to a role

### PR DESCRIPTION
**Pull request by JayaShakthi97**
Screenshot 2023-11-19 at 11 52 15 AM

Screenshot 2023-11-19 at 11 52 29 AM

### Related Issues
- https://github.com/wso2/product-is/issues/17780

### Related PRs
- N/A

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/roles/components/edit-role/edit-role-federated-groups-accordion.tsx [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+                            floated="left"
+                        />
+                        <Typography
+                            className="ml-4"
+                            variant="h6"
+                        >
+                            { identityProvider.name }
+                        </Typography>
+                    </>
+
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-federated-groups-accordion.tsx:253]: Let's remove the redundant line.

File: apps/console/src/features/roles/components/edit-role/edit-role-federated-groups-accordion.tsx [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+                                        disableCloseOnSelect
+                                        options={ selectedGroupsOptions ? selectedGroupsOptions : [] }
+                                        value={ selectedGroupsOptions ? selectedGroupsOptions : [] }
+                                        getOptionLabel={ (group: IdentityProviderGroupInterface) => 
+                                            group.name }
+                                        renderInput={ (params: AutocompleteRenderInputParams) => (
+                                            <TextField
+                                                { ...params }
+                                                placeholder= { t("console:manage.features.roles.edit.groups" + 
+                                                        ".actions.search.placeholder") }
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-federated-groups-accordion.tsx:271]: Let's fix indentation here.

```suggestion
                                                    ".actions.search.placeholder") }
```

File: apps/console/src/features/roles/components/edit-role/edit-role-federated-groups-accordion.tsx [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+                                        disableCloseOnSelect
+                                        loading={ isGroupListFetchRequestLoading || isGroupSearchLoading }
+                                        options={ groupsOptions }
+                                        value={ selectedGroupsOptions ? selectedGroupsOptions : [] }
+                                        getOptionLabel={ (group: IdentityProviderGroupInterface) => group.name }
+                                        renderInput={ (params: AutocompleteRenderInputParams) => (
+                                            <TextField
+                                                { ...params }
+                                                placeholder= { t("console:manage.features.roles.edit.groups" + 
+                                                        ".actions.assign.placeholder") }
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-federated-groups-accordion.tsx:310]: Let's fix indentation here.

File: apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx [hunk 4]
```diff
@@ -71,233 +59,166 @@ export const RoleGroupsList: FunctionComponent<RoleGroupsPropsInterface> = (
...
-    /**
-     * Call mutateGroupListFetchRequest when the group search value changes
-     */
-    useEffect(() => {
-        if ( groupSearchValue ) {
-            mutateGroupListFetchRequest().finally(() => {
-                setGroupSearchLoading(false);
+            applicationData.authenticationSequence.steps.forEach((step: AuthenticationStepInterface) => {
+                step.options.forEach((option: AuthenticatorInterface) => {
+                    const idp: StrictIdentityProviderInterface = filteredList.find(
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx:119]:
```suggestion
                    const idp: StrictIdentityProviderInterface = filteredList?.find(
```

File: apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx [hunk 5]
```diff
@@ -343,203 +264,43 @@ export const RoleGroupsList: FunctionComponent<RoleGroupsPropsInterface> = (
             <Heading subHeading ellipsis as="h6">
                 { t("console:manage.features.roles.edit.groups.subHeading") }
             </Heading>
+            <Heading as="h5">Local Groups</Heading>
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx:267]: Let's add string localization here.

File: apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx [hunk 6]
```diff
@@ -343,203 +264,43 @@ export const RoleGroupsList: FunctionComponent<RoleGroupsPropsInterface> = (
...
+                    onSelectedGroupsListChange={ onSelectedGroupsChange }
+                    isUpdating={ isSubmitting }
+                />
             }
+            {
+                filteredIdpList?.length > 0 && (
+                    <>
+                        <Divider hidden />
+                        <Divider />
+                        <Heading as="h5">External Groups</Heading>
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx:283]: Let's add string localization.

File: apps/console/src/features/roles/components/edit-role/edit-role-federated-groups-accordion.tsx (hunk 1 above)
Comment (JayaShakthi97) [reply apps/console/src/features/roles/components/edit-role/edit-role-federated-groups-accordion.tsx]: Addressed 5ce5a869d09cb34de9eac44ddee5d1053e6006a4

File: apps/console/src/features/roles/components/edit-role/edit-role-federated-groups-accordion.tsx (hunk 2 above)
Comment (JayaShakthi97) [reply apps/console/src/features/roles/components/edit-role/edit-role-federated-groups-accordion.tsx]: Addressed 5ce5a869d09cb34de9eac44ddee5d1053e6006a4

File: apps/console/src/features/roles/components/edit-role/edit-role-federated-groups-accordion.tsx (hunk 3 above)
Comment (JayaShakthi97) [reply apps/console/src/features/roles/components/edit-role/edit-role-federated-groups-accordion.tsx]: Addressed 5ce5a869d09cb34de9eac44ddee5d1053e6006a4

File: apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx (hunk 4 above)
Comment (JayaShakthi97) [reply apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx:119]: Null check is not needed here. `filteredList` won't be undefined or null

File: apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx (hunk 5 above)
Comment (JayaShakthi97) [reply apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx]: Addressed 5ce5a869d09cb34de9eac44ddee5d1053e6006a4

File: apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx (hunk 6 above)
Comment (JayaShakthi97) [reply apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx]: Addressed 5ce5a869d09cb34de9eac44ddee5d1053e6006a4

File: .changeset/real-masks-roll.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+---
+"@wso2is/console": minor
```
Comment (dasuni-30) [inline .changeset/real-masks-roll.md:2]: Shall we bump i18n and myaccount as well?

**Review by dasuni-30 — DISMISSED** [review]

File: .changeset/real-masks-roll.md (hunk 7 above)
Comment (JayaShakthi97) [reply .changeset/real-masks-roll.md]: Addressed fd29c6070019dd2193690e6d85fcae670bd64815

**Review by dasuni-30 — APPROVED** [review]

**Pull request merged**